### PR TITLE
misc stuff

### PR DESCRIPTION
**Pull request by gdementen**

### CODE REVIEW

File: larray/utils.py [hunk 1]
```diff
@@ -438,3 +439,14 @@ def find_closing_chr(s, start=0):
...
+
+
+def float_error_handler_factory(stacklevel):
+    def error_handler(error, flag):
+        if error == 'invalid value':
+            error = 'invalid value (NaN)'
+            extra = ' (this is typically caused by a 0 / 0)'
+        else:
+            extra = ''
+        warnings.warn("{} encountered during operation{}".format(error, extra), RuntimeWarning, stacklevel=stacklevel)
```
Comment (alixdamman) [inline larray/utils.py]: Need an extra withespace after 'operation' in warning.warn

Comment (gdementen) [inline larray/utils.py]: this is intentional (so that if extra is '', we do not get a trailing space).

Comment (gdementen) [inline larray/utils.py]: the "missing" space is within "extra"

Comment (alixdamman) [inline larray/utils.py]: I missed that, sorry. Everything fine then.

File: larray/core.py [hunk 2]
```diff
@@ -4363,6 +4363,14 @@ def __getattr__(self, key):
...
             raise AttributeError("'{}' object has no attribute '{}'".format(self.__class__.__name__, key))
 
+    # needed to make *un*pickling work (because otherwise, __getattr__ is called before .axes exists, which leads to
+    # an infinite recursion)
+    def __getstate__(self):
+        return self.__dict__
+
+    def __setstate__(self, d):
+        self.__dict__ = d
+
```
Comment (alixdamman) [inline larray/core.py]: Are there some tests associated?

Comment (gdementen) [inline larray/core.py]: indirectly, yes. the Session pickling test includes Axis and LArray objects, which means they are pickled and unpickled too.

**Comment by alixdamman** [comment]
Except for the crash with Python 2.7, this LGTM

**Comment by gdementen** [comment]
the fix for 2.7 is coming in a few seconds :)

**Pull request merged**